rust/private/legacy_cc_starlark_api_shim.bzl

```diff
@@ -2,23 +2,33 @@
 Part of --incompatible_disable_legacy_cc_provider migration (https://github.com/bazelbuild/bazel/issues/7036)
 """
 
-def get_libs_for_static_executable(dep):
```

Comment:
nit; not a fan of sorting functions private first

Reply:
Can you explain why? It sorta breaks my brain every time I see private macros declared after they're used. Maybe you can help me with that 😄

Reply:
This is subjective, I prefer seeing top-level things first (main, public functions, interfaces, etc) 

eg. starting with `def do_useful_thing` and ending with `def _do_litany_of_very_specific_things_hopefully_you_dont_need_to_look_in_to`

Reply:
I see, I've reversed the macros. I prefer it the other way but not enough to go against the preferences of a code owner 😄